core/wallet/pem.go

```diff
@@ -133,12 +133,17 @@ func DecryptPEMBlock(b *pem.Block, pwd string) (*pem.Block, error) {
 		return nil, errors.New("x509: malformed DEK-Info header")
 	}
 
-	if mode != "AES-GCM" {
+	// keep compatibility with old PEM encryption
+	var password []byte
+	switch mode {
+	case "AES-GCM":
+		password = getEncryptionKey(pwd, sha1.New)
+	case "AES-256-GCM":
+		password = getEncryptionKey(pwd, sha256.New)
+	default:
```

Comment:
Thanks for heads up dear Rabbit, but we are using SHA-1 only to decrypt in case the user already has a pen file built with an older version of this SDK. So if we complete remove SHA-1, this SDK will stop to work with older encrypted pem files. Note that, the encryption function `EncryptPEMBlock()` only uses SHA-256 for new encryptions.